staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -786,3 +787,12 @@ func AuthorizeClientBearerToken(loopback *restclient.Config, authn *Authenticati
 	tokenAuthorizer := authorizerfactory.NewPrivilegedGroups(user.SystemPrivilegedGroup)
 	authz.Authorizer = authorizerunion.New(tokenAuthorizer, authz.Authorizer)
 }
+
+// IsSelfRequest returns whether req is a self-request on the api server
+func (c *Config) IsSelfRequest(req *http.Request) bool {
+	tok, ok := bearertoken.ExtractBearerToken(req)
+	if ok && (tok == c.LoopbackClientConfig.BearerToken) {
```

Comment:
Yeah, don't do this here. Look for where an authenticator is constructed to match this token instead.